### PR TITLE
refactor: compat with TypeScript 4.6+

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/addon-commerce/utils/get-currency-symbol.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import {TuiCurrency, TuiCurrencyCode} from '@taiga-ui/addon-commerce/enums';
 import {TuiCurrencyVariants} from '@taiga-ui/addon-commerce/types';
 
-export function tuiGetCurrencySymbol(currency: TuiCurrencyVariants): string | null {
+export function tuiGetCurrencySymbol(
+    currency?: TuiCurrencyVariants | null,
```
Comment (nsbarsukov) [inline projects/addon-commerce/utils/get-currency-symbol.ts:5]: I have doubts here. Do we really need it ?

File: projects/addon-table/components/table/tr/tr.component.ts [hunk 2]
```diff
@@ -48,4 +48,9 @@ export class TuiTrComponent<T extends Record<string, any>> {
         @Inject(forwardRef(() => TuiTbodyComponent))
         private readonly body: TuiTbodyComponent<T>,
     ) {}
+
+    @tuiPure
+    keyof$<U>(key: any): keyof U {
```
Comment (nsbarsukov) [inline projects/addon-table/components/table/tr/tr.component.ts:53]:
```suggestion
    $keyof<U>(key: any): keyof U {
```
it is typecasting function (not function which returns Observable).

File: projects/cdk/schematics/ng-add/steps/add-taiga-icons.ts [hunk 3]
```diff
@@ -19,7 +20,7 @@ export function addTaigaIcons(options: Schema): Rule {
 
 export function addTaigaAssetsToAngularJson(options: Schema): Rule {
     return updateWorkspace(workspace => {
-        const project = getProject(options, workspace);
+        const project = getProject(options, workspace as unknown as WorkspaceDefinition);
```
Comment (nsbarsukov) [inline projects/cdk/schematics/ng-add/steps/add-taiga-icons.ts:23]: Did you update (locally) `@schematics/angular/utility/workspace` after angular update ?

Probably, we dont need typecasting here (if `@schematics/angular` has the same version with `@angular-devkit`).

**Review by nsbarsukov — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/addon-commerce/utils/get-currency-symbol.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-commerce/utils/get-currency-symbol.ts]: fixed

File: projects/addon-table/components/table/tr/tr.component.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-table/components/table/tr/tr.component.ts]: fixed

File: projects/cdk/schematics/ng-add/steps/add-taiga-icons.ts (hunk 3 above)
Comment (splincode) [reply projects/cdk/schematics/ng-add/steps/add-taiga-icons.ts]: fixed

**Pull request merged**